plugins/modules/authorized_key.py

```diff
@@ -618,9 +618,12 @@ def enforce_state(module, params):
     # for 'exclusive', make sure keys are written in the order the new keys were
     if state == "present" and exclusive:
         to_remove = frozenset(existing_keys).difference(keys_to_exist)
-        for key in to_remove:
-            del existing_keys[key]
-            do_write = True
+        if len(to_remove) != 0:
+            for key in to_remove:
+                del existing_keys[key]
+                do_write = True
+        else:
+            do_write = False
```

Comment:
hi @Bilanda thanks for your contribution

I think this could open another issue. What I mean is:
- `do_write` has been initialized to `False`
- now you override it to `False` again
- what happened between ?
I guess that if there is an issue somewhere, it could rather be found after line 594 and before line 615, not here

In this fix, `do_write` is ***forced*** to `False` if we want the key to be `present` and `exclusive`, and if there is no key to remove. So, what in the case there is nothing to remove but something to add (the key we want to be present and exclusive and that is not already there) ? We get `do_write=False`, and I'm not sure this is want we want.

There is something unclear for me here, or in the description, or both. As mentioned first by @aminvakil , this is related to the `exclusive` option. Could you please provide a sample play that produces the error without your change, and that doesn't produce the error with your change ? It would be the best starting point for adding integration tests :)

Reply:
Looking at the logic in the code, `do_write` is initialized to `False`, and possibly overridden to `True` in a loop, that means it is not expected to be reset to `False`, neither in the loop itself, nor at any time, without breaking the logic: `do_write` is a kind of *no-return-switch*, that once toggled to the opposite of its initial value, never goes back. Or think of it as an ORed sequence of booleans: (false or false or false or true or false or false) is `true`.

Many modules make use of this logic for the `changed` variable that is returned in their results.

Reply:
Hello @quidame, thanks a lot for your support.
As mentionned before, i tried with @aminvakil's playbook and i still have the issue.
I used this kind of task from my side where i have the problem : 

`- name: Install root SSH authorized keys
  authorized_key:
    state: present
    user: root
    manage_dir: yes
    exclusive: yes
    key: "{{ lookup('template', 'root/.ssh/authorized_keys.j2') }}"`

But whenever i run the playbook, even if there is absolutely no change in the deployed keys, the task return _changed_
I can run playbook in verbose mode if this can help.
Thanks again